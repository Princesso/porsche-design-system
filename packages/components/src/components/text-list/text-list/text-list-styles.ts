import type { Theme } from '../../../types';
import { buildSlottedStyles, getCss } from '../../../utils';
import { addImportantToEachRule, getBaseSlottedStyles, getThemedColors, hostHiddenStyles } from '../../../styles';
import { textSmallStyle } from '@porsche-design-system/utilities-v2';

export const getComponentCss = (theme: Theme): string => {
  return getCss({
    '@global': {
      ':host': {
        display: 'block',
        ...addImportantToEachRule({
          counterReset: 'section',
          ...hostHiddenStyles,
        }),
      },
      '[role]': {
        display: 'block',
        padding: 0,
        margin: 0,
        color: getThemedColors(theme).primaryColor,
        ...textSmallStyle,
      },
    },
  });
};

export const getSlottedCss = (host: HTMLElement): string => {
  return getCss(buildSlottedStyles(host, getBaseSlottedStyles({ withDarkTheme: true })));
};
