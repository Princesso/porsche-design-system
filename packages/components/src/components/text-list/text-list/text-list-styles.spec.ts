import { getComponentCss } from './text-list-styles';
import type { Theme } from '../../../types';

describe('getComponentCss()', () => {
  it.each<Theme>(['light', 'dark'])('should return correct css for theme: %s', (theme) => {
    expect(getComponentCss(theme)).toMatchSnapshot();
  });
});
