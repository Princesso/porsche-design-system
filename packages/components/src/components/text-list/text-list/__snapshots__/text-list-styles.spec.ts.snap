// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getComponentCss() should return correct css for listType: ordered, orderType: alphabetically and theme: dark 1`] = `
":host {
  display: block;
}
:host([hidden]) {
  display: none !important;
}
ol,ul {
  font: normal normal 400 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
  color: #FBFCFF;
  margin: 0;
  padding: var(--p-internal-text-list-padding-top,0) 0 var(--p-internal-text-list-padding-bottom,0) 1.5rem;
  list-style-type: lower-latin;
}
::slotted(*) {
  --p-internal-text-list-padding-top: 4px;
  --p-internal-text-list-padding-bottom: 16px;
  --p-internal-text-list-padding-left: .625rem;
  --p-internal-text-list-list-style-type: "–";
}"
`;

exports[`getComponentCss() should return correct css for listType: ordered, orderType: alphabetically and theme: light 1`] = `
":host {
  display: block;
}
:host([hidden]) {
  display: none !important;
}
ol,ul {
  font: normal normal 400 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
  color: #010205;
  margin: 0;
  padding: var(--p-internal-text-list-padding-top,0) 0 var(--p-internal-text-list-padding-bottom,0) 1.5rem;
  list-style-type: lower-latin;
}
::slotted(*) {
  --p-internal-text-list-padding-top: 4px;
  --p-internal-text-list-padding-bottom: 16px;
  --p-internal-text-list-padding-left: .625rem;
  --p-internal-text-list-list-style-type: "–";
}"
`;

exports[`getComponentCss() should return correct css for listType: ordered, orderType: numbered and theme: dark 1`] = `
":host {
  display: block;
}
:host([hidden]) {
  display: none !important;
}
ol,ul {
  font: normal normal 400 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
  color: #FBFCFF;
  margin: 0;
  padding: var(--p-internal-text-list-padding-top,0) 0 var(--p-internal-text-list-padding-bottom,0) 1.5rem;
  list-style-type: decimal;
}
::slotted(*) {
  --p-internal-text-list-padding-top: 4px;
  --p-internal-text-list-padding-bottom: 16px;
  --p-internal-text-list-padding-left: .625rem;
  --p-internal-text-list-list-style-type: "–";
}"
`;

exports[`getComponentCss() should return correct css for listType: ordered, orderType: numbered and theme: light 1`] = `
":host {
  display: block;
}
:host([hidden]) {
  display: none !important;
}
ol,ul {
  font: normal normal 400 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
  color: #010205;
  margin: 0;
  padding: var(--p-internal-text-list-padding-top,0) 0 var(--p-internal-text-list-padding-bottom,0) 1.5rem;
  list-style-type: decimal;
}
::slotted(*) {
  --p-internal-text-list-padding-top: 4px;
  --p-internal-text-list-padding-bottom: 16px;
  --p-internal-text-list-padding-left: .625rem;
  --p-internal-text-list-list-style-type: "–";
}"
`;

exports[`getComponentCss() should return correct css for listType: unordered, orderType: numbered and theme: dark 1`] = `
":host {
  display: block;
}
:host([hidden]) {
  display: none !important;
}
ol,ul {
  font: normal normal 400 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
  color: #FBFCFF;
  margin: 0;
  padding: var(--p-internal-text-list-padding-top,0) 0 var(--p-internal-text-list-padding-bottom,0) var(--p-internal-text-list-padding-left,.375rem);
  list-style-type: var(--p-internal-text-list-list-style-type,'•');
}
::slotted(*) {
  --p-internal-text-list-padding-top: 4px;
  --p-internal-text-list-padding-bottom: 16px;
  --p-internal-text-list-padding-left: .625rem;
  --p-internal-text-list-list-style-type: "–";
}"
`;

exports[`getComponentCss() should return correct css for listType: unordered, orderType: numbered and theme: light 1`] = `
":host {
  display: block;
}
:host([hidden]) {
  display: none !important;
}
ol,ul {
  font: normal normal 400 1rem/calc(6px + 2.125ex) 'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;
  overflow-wrap: break-word;
  hyphens: auto;
  color: #010205;
  margin: 0;
  padding: var(--p-internal-text-list-padding-top,0) 0 var(--p-internal-text-list-padding-bottom,0) var(--p-internal-text-list-padding-left,.375rem);
  list-style-type: var(--p-internal-text-list-list-style-type,'•');
}
::slotted(*) {
  --p-internal-text-list-padding-top: 4px;
  --p-internal-text-list-padding-bottom: 16px;
  --p-internal-text-list-padding-left: .625rem;
  --p-internal-text-list-list-style-type: "–";
}"
`;
